### PR TITLE
aws - ec2 - set-metadata-access - include instance tags option

### PR DESCRIPTION
**Pull request by farzanv**
This change has been added to enable Instancemetadata access on AWS.ec2 instances.

### CODE REVIEW

File: c7n/resources/ec2.py [hunk 1]
```diff
@@ -1137,24 +1137,36 @@ class SetMetadataServerAccess(BaseAction):
...
 
     schema = type_schema(
         'set-metadata-access',
         anyOf=[{'required': ['endpoint']},
                {'required': ['tokens']},
+               {'required': ['metadatatags']},
                {'required': ['hop-limit']}],
         **{'endpoint': {'enum': AllowedValues['HttpEndpoint']},
            'tokens': {'enum': AllowedValues['HttpTokens']},
+           'metadatatags': {'enum': AllowedValues['InstanceMetadataTags']},
```
Comment (kapilt) [inline c7n/resources/ec2.py:1169]: I think we want the hype for word separator to match the hop-limit option.

**Comment by kapilt** [comment]
probably need to rebase on trunk to get around the poetry.lock issues / conflicts

**Review by kapilt — APPROVED** [review]
thanks, lgtm

File: tests/test_ec2_InstanceTags.py [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+            output.getvalue(),
+            ('set-metadata-access implicitly filtered 1 of 2 resources '
+             'key:MetadataOptions.HttpTokens on optional\n'))
+
+
+class TestSetMetadataTags(BaseTest):
+
+    def test_set_metadata_server(self):
+        output = self.capture_logging('custodian.actions')
+        session_factory = self.replay_flight_data('test_ec2_set_md_access')
```
Comment (kapilt) [inline tests/test_ec2_InstanceTags.py:152]: this needs to be a separate directory, else its overwriting other tests.

File: tests/terraform/TestSetMetadataTags/main.tf [hunk 3]
```diff
@@ -0,0 +1,19 @@
+resource "aws_instance" "no_protection" {
```
Comment (kapilt) [inline tests/terraform/TestSetMetadataTags/main.tf:1]: this only needs the one instance for metadata tags

Comment (farzanv) [reply tests/terraform/TestSetMetadataTags/main.tf:1]: removed, Thanks

File: tests/test_ec2_InstanceTags.py (hunk 2 above)
Comment (farzanv) [reply tests/test_ec2_InstanceTags.py]: The file has been moved to tests/ec2_metadatatags/test_ec2_InstanceTags.py

File: pytest_terraform/__init__.py [hunk 4]
```diff
@@ -0,0 +1,18 @@
+# Copyright 2020 Kapil Thangavelu
```
Comment (kapilt) [inline pytest_terraform/__init__.py:1]: none of the code in this package should be in this pr. its a separate repository/dependency.

**Pull request merged**